gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -199,24 +223,36 @@ public static Builder newBuilder() {
   }
 
   public static final class Builder {
+    private Runtime runtime;
     private ExecutorProvider executorProvider;
     private HeaderProvider headerProvider;
     private String endpoint;
     @Nullable private Integer maxInboundMessageSize;
     @Nullable private Duration keepAliveTime;
     @Nullable private Duration keepAliveTimeout;
     @Nullable private Boolean keepAliveWithoutCalls;
+    private int poolSize;
 
-    private Builder() {}
+    private Builder() {
+      runtime = Runtime.getRuntime();
+    }
```

Comment:
It seems a little odd to inject `Runtime`. I think we should inject the number of available processors instead? Then we don't have to mock, just inject a number.

Reply:
yep, thats much cleaner